src/util/address.rs

```diff
@@ -789,4 +823,60 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_valid_networks() {
+        let legacy_payload = &[
+            Payload::PubkeyHash(PubkeyHash::default()),
+            Payload::ScriptHash(ScriptHash::default())
+        ];
+        let segwit_payload = &[
+            Payload::WitnessProgram {
+                version: Default::default(),
```

Comment:
Correct me if I'm wrong, but we need to test different segwit versions here, aren't we?

This implies the question if Signet P2TR address is equivalent to Testnet P2TR address, which before the November is not deployed to the network...

Reply:
P2TR addresses are already valid (or they would be if there wasn't the bech32m problem) and would ideally follow the same conventions as other bech32 addresses. This means we can certainly not say that they aren't compatible, because that might limit interoperability (a wallet doesn't need to be aware of the destination address version's semantics with bech32 addresses and should "just work").

I also think that practically it will be easier to tighten the rules down the road should anything change.

A compromise would be to have three states: compatible, incompatible and unknown. That would allow the user to decide what to do.